libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -404,6 +418,55 @@ static bool _createKeepAliveOperation( const IotMqttNetworkInfo_t * pNetworkInfo
 
 /*-----------------------------------------------------------*/
 
+static bool _initializeNetworkConnection( const IotMqttNetworkInfo_t * pNetworkInfo,
+                                          void ** pNetworkConnection,
+                                          bool * pCreatedNetworkConnection )
+{
+    IOT_FUNCTION_ENTRY( bool, true );
+    bool networkStatus = false;
```

Comment:
Is `bool` the best type for this given that `pNetworkInterface->create()` returns an `IotNetworkError_t`?

Reply:
It is not, good catch.  Reworked.